selftests/vmimage/tests/unit_test_runner_vmimage.py

```diff
@@ -0,0 +1,127 @@
+import logging
+import sys
+import unittest
+
+from avocado.core.nrunner.runnable import Runnable
+from avocado.plugins.runners.vmimage import VMImageRunner
+from avocado.utils import path
+from selftests.vmimage.tests.utils.network import is_network_available
+
+RUNNER = f"{sys.executable} -m avocado.plugins.runners.vmimage"
+
+
+def missing_binary(binary):
+    try:
+        path.find_command(binary)
+        return False
+    except path.CmdNotFoundError:
+        return True
+
+
+@unittest.skipUnless(
+    is_network_available(),
+    "Network connectivity required to run these tests",
+)
+@unittest.skipIf(
+    missing_binary("qemu-img"),
+    "QEMU disk image utility is required by the vmimage utility ",
+)
+class VMImageRunnerTest(unittest.TestCase):
+    def setUp(self):
+        self.log = logging.getLogger("avocado.test")
+
+    def test_vmimage_basic(self):
+        """Test the vmimage runner using the Python API directly."""
+        runnable = Runnable(
+            "vmimage",
+            "",
+            kwargs={
+                "provider": "Fedora",
+                "version": "41",
+                "arch": "x86_64",
+            },
+            config={"debug": True},
+        )
+        runner = VMImageRunner()
+        try:
+            print("Running vmimage runner...")
```

Comment:
Don't use prints in tests, use only asserts and assert error messages for failure explanations. If you need to log something use python logging, but IMO in here the logging is not needed.